internal/podman/client.go

```diff
@@ -201,8 +217,51 @@ func (o *client) machineInit(ctx context.Context) error {
 		return nil
 	}
 
-	_, err = o.runPodman(ctx, "machine", "init")
-	return err
+	if _, err = o.runPodman(ctx, newMachineInitArgs()...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func newMachineInitArgs() []string {
+	args := []string{"machine", "init"}
+	memory, _ := mem.VirtualMemory()
+	defaultMachineMemoryUint64, err := strconv.ParseUint(defaultMachineMemory, 10, 64)
+	if err != nil {
+		_, _ = log.Warning(err)
+	}
+
+	if memory.Available > defaultMachineMemoryUint64 {
+		args = append(args, "--memory", defaultMachineMemory)
+	} else {
+		_, _ = log.Warningf(notEnoughMemoryAvailable, memory.Available, defaultMachineMemory)
+	}
+
+	cores, err := cpu.Counts(false)
```

Comment:
just for my curiosity what is the difference between this and https://pkg.go.dev/runtime#NumCPU ?

Reply:
> NumCPU returns the number of logical CPUs usable by the current process.
sure, my understanding is that the CPU usable by the current GO process can be different than the physical system cpus

Reply:
@andreaangiolillo do you mind double checking if you have different results? my understanding is that usable for Go is the same as present on System, it should be all virtual CPUs

Reply:
I was not able to find a scenario where `NumCPU` or `cpu.Counts` have different results. Defaulting back to NumCPU

Reply:
updated